app/lib/katello/resources/discovery/yum_discovery.rb

```diff
@@ -0,0 +1,73 @@
+module Katello
+  class YumDiscovery < RepoDiscovery
+    attr_reader :found, :crawled, :to_follow
+    def initialize(url, crawled = [], found = [], to_follow = [],
+                   upstream_credentials_and_search = {
+                     upstream_username: nil,
+                     upstream_password: nil
+                   })
+      @uri = uri(url)
+      @upstream_username = upstream_credentials_and_search[:upstream_username].empty? ? nil : upstream_credentials_and_search[:upstream_username]
+      @upstream_password = upstream_credentials_and_search[:upstream_password].empty? ? nil : upstream_credentials_and_search[:upstream_password]
```

Comment:
I know it was already this way, but isn't this equivalent?
```suggestion
      @upstream_username = upstream_credentials_and_search[:upstream_username].presence
      @upstream_password = upstream_credentials_and_search[:upstream_password].presence
```